packages/pyright-internal/src/analyzer/typeNameUtils.ts

```diff
@@ -0,0 +1,90 @@
+import { getEnclosingClass } from './parseTreeUtils';
+import { Scope } from './scope';
+import { SymbolTable } from './symbol';
+import { TypeEvaluator } from './typeEvaluatorTypes';
+import { ClassType, isClass, isModule, ModuleType, TypeBase } from './types';
+
+function handleSymbolTable(evaluator: TypeEvaluator, type: ClassType | ModuleType, symbolTable: SymbolTable): string[] {
+    const getFullName = (t: ClassType | ModuleType) => (isClass(t) ? t.shared.fullName : t.priv.moduleName);
+    const fullName = getFullName(type);
+
+    // The symbols in symbolTable whose type is a non-instance class or a module
+    const symbols: { type: ClassType | ModuleType; localName: string; fullName: string }[] = [];
+    symbolTable.forEach((symbol, symbolName) => {
+        const symbolType = evaluator.getEffectiveTypeOfSymbol(symbol);
+        if ((isClass(symbolType) && !TypeBase.isInstance(symbolType)) || isModule(symbolType)) {
+            symbols.push({ type: symbolType, localName: symbolName, fullName: getFullName(symbolType) });
+        }
+    });
+
+    // If there are symbols whose type has the same full name as `type`, return these
+    const exactMatches = symbols.filter((symbolInfo) => symbolInfo.fullName === fullName);
+    if (exactMatches.length > 0) return exactMatches.map((symbolInfo) => symbolInfo.localName);
+
+    // If there are no full matches, try to find types whose name is a prefix to `type`’s full name
+    // and then recursively try to find a symbol with `type`’s full type.
+    return symbols
+        .filter((symbolInfo) => fullName.startsWith(symbolInfo.fullName))
+        .flatMap((symbolInfo) =>
+            getLocalTypeNameRec(evaluator, type, symbolInfo.type).map((name) => `${symbolInfo.localName}.${name}`)
+        );
+}
+
+function getLocalTypeNameRec(
+    evaluator: TypeEvaluator,
+    type: ClassType | ModuleType,
+    symbolType: ClassType | ModuleType
+) {
+    return handleSymbolTable(
+        evaluator,
+        type,
+        isClass(symbolType) ? ClassType.getSymbolTable(symbolType) : symbolType.priv.fields
+    );
+}
```

Comment:
all this function does is call `handleSymbolTable`, i think it makes sense to inline it